CSharpFunctionalExtensions/Result/Extensions/TapAsyncBoth.cs

```diff
@@ -69,5 +69,35 @@ public static async Task<Result<T>> Tap<T>(this Task<Result<T>> resultTask, Func
 
             return result;
         }
+
+        /// <summary>
+        ///     Executes the given action if the calling result is a success. Returns the calling result on success.
+        /// </summary>
+        public static async Task<Result> Tap<T>(this Task<Result<T>> resultTask, Func<T, Task<Result>> func)
+        {
+            Result<T> result = await resultTask.DefaultAwait();
+            var funcResult = await result.Bind(func).DefaultAwait();
+            return funcResult.Map(() => result.Value);
+        }
+
+        /// <summary>
+        ///     Executes the given action if the calling result is a success. Returns the calling result on success.
+        /// </summary>
+        public static async Task<Result<T>> Tap<T, K>(this Task<Result<T>> resultTask, Func<T, Task<Result<K>>> func)
+        {
+            Result<T> result = await resultTask.DefaultAwait();
+            var funcResult = await result.Bind(func).DefaultAwait();
+            return funcResult.Map(_ => result.Value);
+        }
```

Comment:
See comment on `TapAsyncRight`. Couldn't we do this?

```csharp
public static async Task<Result<T>> Tap<T, K>(this Task<Result<T>> resultTask, Func<T, Task<Result<K>>> func)
{
    Result<T> result = await resultTask.DefaultAwait();
    return await result.Bind(func).Map(_ => result.Value).DefaultAwait();
}
```

Reply:
Updated.